tools/mpremote/mpremote/commands.py

```diff
@@ -334,6 +334,49 @@ def do_filesystem_recursive_rm(state, path, args):
             print(f"removed: '{path}'")
 
 
+def human_size(size, decimal_places=1):
+    for unit in ['B', 'K', 'M', 'G', 'T']:
+        if size < 1024.0 or unit == 'T':
+            break
+        size /= 1024.0
+    return f"{size:.{decimal_places}f} {unit}" if unit != 'B' else f"{int(size)} {unit}"
+
+
+def do_filesystem_tree(state, path, args):
+    """Print a tree of the device's filesystem starting at path."""
+    connectors = ("├── ", "└── ")
+
+    # connectors = ("|-- ", "+-- ")
+    def _tree_recursive(path, prefix=""):
+        entries = state.transport.fs_listdir(path)
+        entries.sort(key=lambda e: e.name)
+        for i, entry in enumerate(entries):
+            connector = connectors[1] if i == len(entries) - 1 else connectors[0]
+            is_dir = entry.st_mode & 0x4000  # Directory
+            if args.size and not is_dir:
+                print(f"{prefix}{connector}[{entry.st_size:>12}] {entry.name}")
+            elif args.human and not is_dir:
+                print(f"{prefix}{connector}[{human_size(entry.st_size):>8s}] {entry.name}")
```

Comment:
As above, the width here could probably be less.  I think 6 is enough to fit everything (without a space between the number and unit, see above).